### ISSUE TITLE
1.0.5 Error dialog is thrown when repeatedly performing a play operation with a start point configured

### ISSUE THREAD

**Issue by worblyhead** [Nornec/Midinous-Issues#8]
In the 1.0.5 Beta I was going through the various steps of the initial tutorial when I got an error dialog for performing Play operations (I was using the spacebar to initiate play).  In this case there is indeed a Start point configured.  

1) Open Midinous Beta 1.0.5 and start the tutorial

2) Scroll to the part of the tutorial discussing Start points

3) Repeatedly hit spacebar (or Play control) and notice that every once in awhile an error dialog is thrown (please seen attached screencap) .

Not sure if this is timing event/control focus related or not.  Pressing the spacebar works most of the time, though every once in awhile the error dialog is thrown.

<img width="816" alt="Midinous_StartBehavior_1" src="https://user-images.githubusercontent.com/115739648/195728359-a654bc11-f4c2-4e47-83b0-74e3cc2e2250.png">

**Comment by worblyhead** [comment]
The more I try to reproduce this the more it appears to be simply a timing issue.  If play is repeated faster than some delta t than the dialog results.  I tried to use just 1 Midi primary start point and extend the duration value and couldn't get the issue to reproduce.

**Comment by Nornec** [comment]
This is an interesting one -- it may be a race condition between tasks. I'll look into this and debug appropriately.

**Comment by Nornec** [comment]
I cannot reproduce this. The error may have been related to the object ID bug that was causing crashes before. In the next release, we will have to see if this still happens.

**Comment by worblyhead** [comment]
I just updated to Beta 1.1.0 and this issue persists.  I simply placed a start MIDI point and consecutively hit the space bar to initiate play over and over again.  Didn't take too many clicks to get the error dialog to be thrown.

**Comment by Nornec** [comment]
That's very strange. I was mashing play and space for a few minutes
yesterday in multiple files without getting your error message. What is
your system info? This may call for a screenshare or a video so i can see
it in action

**Comment by worblyhead** [comment]
[image: image.png]

**Comment by Nornec** [comment]
I don't think your image upload worked. But a video would be better

**Comment by worblyhead** [comment]
![image](https://user-images.githubusercontent.com/115739648/196719484-9646948b-bef2-4890-853d-d00db0b8458f.png)

**Comment by worblyhead** [comment]
One other thing.  I went through changing the Duration value of the start point as in the screenshot below.  A value of 1.5 almost immediately gets the dialog to appear more so than any other value that I tried.  For the default value of 1, you can get the behavior to reproduce, but since 1.1.0 you have to click Start (i.e. the space bar) a lot more (or faster) than in 1.0.5 FWIW.


![image](https://user-images.githubusercontent.com/115739648/197344562-82d0edd9-0f83-4c6e-bf40-a61d24b7c0ff.png)

**Comment by Nornec** [comment]
This appears to have been caused by two things
1) Improper error dialog logic
2) Race conditions with the threads used for playing.

This issue flew under the radar for me because my test bench is extremely performant. Only when I increased the # of objects to a point where I could start to see performance issues did this problem appear for me.

I believe this to be solved in the next beta release for 1.1.0.
